### ISSUE TITLE
Bug: [class-methods-use-this] Rule shouldn't report methods overriden from parent class

### ISSUE THREAD

**Issue by marekdedic** [typescript-eslint/typescript-eslint#8481]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=FAYwNghgzlAEBC0CmsDeti1gBwK4CMwBLEWAEwHsBlAF1wDN6AKASjUy1hAoDsoKwSAHRgKAcyY0AFkSgsOAX2BLQkGLAAiSAE5EAbkjKwkADxpIeZOIigpUHPIRLlqdRqzRKFQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6M%2BAQ2WVoFtE%2BABYB7ACa9YyRLWGVUGSImjQR0SODABfEFqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Base { 
  public doStuff() {
    console.log(this)
  }
}

class Derived extends Base {
  public doStuff() {}
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/class-methods-use-this": "error"
  }
}
```

### tsconfig

```jsonc
{}
```

### Expected Result

No error for `Derived.doStuff()`

### Actual Result

`Expected 'this' to be used by class method 'doStuff'. 8:3 - 8:17`

### Additional Info

I believe the function shouldn't be reported, because it overrides an inherited function and as such cannot be converted to a static function.

**Comment by auvred** [comment]
`class-methods-use-this` has [`"ignoreOverrideMethods"`](https://typescript-eslint.io/rules/class-methods-use-this/#ignoreoverridemethods) option. I guess it should do the trick.

[Playground](https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=FAYwNghgzlAEBC0CmsDeti1gBwK4CMwBLEWAEwHsBlAF1wDN6AKASjUy1hAoDsoKwSAHRgKAcyY0AFkSgsOAX2BLQkGLAAiSAE5EAbkjKwkADxpIeZOIigpUHPIRKwKB7brIpKtBszaolBSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6M%2BAQ2WVoFtE%2BABYB7ACa9YyRLWGVUGANqRE0aCOiQANJkiUA5kw2IA8gDdVVMYgCyg0RPRh8cRAF8AuuDBuQboA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by marekdedic** [comment]
Ugh, I am apparently blind, I was looking for it on that very page :facepalm: 

Sorry, closing...
